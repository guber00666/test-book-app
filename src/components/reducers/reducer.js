
export default function reducer(state, action) {

    switch(action.type) {
    case("setcriteriaValue") :
    break;

    case("setfiltervalue") :
    break;

    default : 
    return state;
    }

}