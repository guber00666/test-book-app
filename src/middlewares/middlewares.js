import {booksFetchDataSuccess, booksHasErrored, booksIsLoading} from "../actions/actions";

export const logger = store => next => action => {
    console.log('dispatching', action);
    let result = next(action);
    console.log('next state', store.getState());
    return result
  };
  
export const crashReporter = store => next => action => {
    try {
      return next(action)
    } catch (err) {
      console.error('Caught an exception!', err);
      Raven.captureException(err, {
        extra: {
          action,
          state: store.getState()
        }
      });
      throw err
    }
  };

export function booksFetchData(url) {
    return (dispatch) => {

        fetch(url)
            .then((response) => {
                if (!response.ok) {
                    throw Error(response.statusText);
                }

                dispatch(booksIsLoading(false));

                return response;
            })
            .then((response) => response.json())
            .then((items) => dispatch(booksFetchDataSuccess(items)))
            .catch(() => dispatch(booksHasErrored(true)));
    };
}


  