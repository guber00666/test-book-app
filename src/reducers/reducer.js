import { criteriaOptions } from '../services/books-service';

export const initialState = {
    criteriaValue: criteriaOptions[0].value,
    filterValue: '',
    paginatorValue: 1,
};

export default function reducer(state = initialState, action) {
    switch (action.type) {
        case "SET_CRITERIA":
            return {
                ...state,
                criteriaValue: action.criteriaValue
            };

        case "SET_FILTER":
            return {
                ...state,
                filterValue: action.filterValue
            };

        case "SET_PAGE":
            return {
                ...state,
                paginatorValue: action.paginatorValue,
            };

        default:
            return state;
    }


}
export const mapStateToProps = (state) => {
    return {
        criteriaValue: state.criteriaValue,
        filterValue: state.filterValue,
        paginatorValue: state.paginatorValue
    }
}






