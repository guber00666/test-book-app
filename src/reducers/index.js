import reducer, { initialState } from './reducer';

export { initialState };
export default reducer;
