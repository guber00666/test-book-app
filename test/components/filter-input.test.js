import React from "react";
import Adapter from "enzyme-adapter-react-16/build";
import FilterInput from '../../src/components/filter-input';
import { createStore, applyMiddleware } from 'redux';
import {Provider} from "react-redux";
import {initialState} from '../../src/reducers/reducer'
import 'jest-styled-components';

import {
    mapStateToFilterProps,
    mapDispatchToFilterProps
} from '../../src/actions/actions'
import reducer from "../../src/reducers";
import {crashReporter, logger} from "../../src/middlewares/middlewares";
import thunk from "redux-thunk";


configure({ adapter: new Adapter() });

const store = createStore(reducer,
    applyMiddleware(logger, crashReporter, thunk));


describe("<FilterInput /> ", () => {
    const wrapperFilter = mount(
        <Provider store={store}>
            <FilterInput />
        </Provider>
    );

    it('Filterinput should render self and other imports', () => {
        expect(wrapperFilter).toMatchSnapshot();
    });

    it('check input correct working', () => {
        const event = {target: {value: '1'}};
        expect(wrapperFilter.find('input').props().value).toEqual('');
        wrapperFilter.find('input').simulate('change', event);
        expect(wrapperFilter.find('input').props().value).toEqual('1');
    });

    it('filterValue should show previously rolled value', () => {
        expect(mapStateToFilterProps(initialState).filterValue).toEqual("");
    });

    it('filterValue should roll the dice again when onChange work out', () => {
        const dispatch = jest.fn();
        mapDispatchToFilterProps(dispatch).setFilterValue();
        expect(dispatch.mock.calls[0][0]).toEqual({  type: 'SET_FILTER' });
    });

});