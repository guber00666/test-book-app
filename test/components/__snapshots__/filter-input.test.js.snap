// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterInput Component should be render on FilterInput 1`] = `
<div
  className="search-panel"
>
  <div
    className="input-group mb-3"
  />
  <styled.div>
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="type for search"
    />
  </styled.div>
</div>
`;
